src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -40,39 +40,83 @@
+     * calculated by finding the total size (in bytes) of all the files under the given input path, then
+     * dividing by the default split size (10MB).
+     * @param ctx the Spark context
+     * @param filePath the path to load files from
+     * @param defaultParallelism the default parallelism to use if the size of the input cannot be determined
+     * @return the recommended parallelism, or <code>defaultParallelism</code> if the size of the input
+     * cannot be determined
      */
-    public JavaRDD<GATKRead> getParallelReads(final String readFileName, final String referencePath, final List<SimpleInterval> intervals) {
-        return getParallelReads(readFileName, referencePath, intervals, DEFAULT_SPLIT_SIZE);
+    public static int getRecommendedParallelism(final JavaSparkContext ctx, final String filePath, final int defaultParallelism) {
+        return getRecommendedParallelism(ctx, filePath, defaultParallelism, DEFAULT_SPLIT_SIZE);
+    }
+
+    /**
+     * Return the recommended parallelism for a pipeline processing the given input path. The parallelism is
+     * calculated by finding the total size (in bytes) of all the files under the given input path, then
+     * dividing by the default split size.
+     * @param ctx the Spark context
+     * @param filePath the path to load files from
+     * @param defaultParallelism the default parallelism to use if the size of the input cannot be determined
+     * @param splitSize maximum bytes of bam file to read into a single partition, increasing this will result in lower parallelism
+     * @return the recommended parallelism, or <code>defaultParallelism</code> if the size of the input
+     * cannot be determined
+     */
+    public static int getRecommendedParallelism(final JavaSparkContext ctx, final String filePath, final int defaultParallelism, int splitSize) {
+        long size = getInputSizeInBytes(ctx, filePath);
+        return size == -1L ? defaultParallelism : (int) (size / splitSize);
     }
 
     /**
-     * Loads Reads using Hadoop-BAM. For local files, bam must have the fully-qualified path,
+     * Loads Reads using Hadoop-BAM. For local files, readFileName must have the fully-qualified path,
      * i.e., file:///path/to/bam.bam.
      * @param readFileName file to load
      * @param referencePath Reference path or null if not available. Reference is required for CRAM files.
      * @param intervals intervals of reads to include.
-     * @param splitSize maximum bytes of bam file to read into a single partition, increasing this will result in fewer partitions
      * @return RDD of (SAMRecord-backed) GATKReads from the file.
      */
-    public JavaRDD<GATKRead> getParallelReads(final String readFileName, final String referencePath, final List<SimpleInterval> intervals, long splitSize) {
+    public JavaRDD<GATKRead> getParallelReads(final String readFileName, final String referencePath, final List<SimpleInterval> intervals) {
```

Comment:
Restore the overload of `getParallelReads` that takes a `splitSize` arg to allow manual control when we run into memory issues with a tool.